internal/cli/atlas/backup/snapshots/delete.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snapshots
+
+import (
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type DeleteOpts struct {
+	cli.GlobalOpts
+	*cli.DeleteOpts
+	clusterName string
+	store       store.SnapshotsDeleter
+}
+
+func (opts *DeleteOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *DeleteOpts) Run() error {
+	return opts.Delete(opts.store.DeleteSnapshot, opts.ConfigProjectID(), opts.clusterName)
+}
+
+// mongocli atlas snapshot(s) delete <snapshotId> --force --clusterName [--projectId projectId]
+func DeleteBuilder() *cobra.Command {
+	opts := &DeleteOpts{
+		DeleteOpts: cli.NewDeleteOpts("Snapshot '%s' deleted\n", "Snapshot not deleted"),
+	}
+	cmd := &cobra.Command{
+		Use:     "delete <snapshotId>",
+		Aliases: []string{"rm"},
+		Short:   deleteSnapshot,
+		Args:    cobra.ExactArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := opts.PreRunE(opts.initStore); err != nil {
+				return err
+			}
+			opts.Entry = args[0]
+			return opts.Prompt()
```

Comment:
I think we need to move this to the `RunE` block, cobra validates missing flags after the `PreRunE` so we will ask this even if `ClusterName` is missing